core/src/services/analytics/sboms/provider.rs

```diff
@@ -0,0 +1,84 @@
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::sync::Arc;
+use platform::mongodb::{Service, Store};
+use crate::entities::tasks::{Task, TaskStatus};
+use crate::Error;
+use crate::services::analytics::sboms::service::AnalyticService;
+use crate::services::tasks::TaskProvider;
+use chrono::Utc;
+use async_trait::async_trait;
+
+/// AnalyticExecutionTask to add errors to the data store
+#[derive(Debug)]
+pub struct AnalyticExecutionTask {
+    service: AnalyticService
+}
+
+impl AnalyticExecutionTask {
+
+    /// Creates a new AnalyticExecutionTask
+    pub fn new(service: AnalyticService) -> Self {
+        Self {
+            service
+        }
+    }
+}
+
+impl Service<Task> for AnalyticExecutionTask {
+    fn store(&self) -> Arc<Store> {
+        self.service.store.clone()
+    }
+}
+
+#[async_trait]
+impl TaskProvider for AnalyticExecutionTask {
+
+    async fn run(&mut self, task: &mut Task) -> Result<HashMap<String, String>, Error> {
+
+        task.start = Utc::now();
+
+        let mut errors = HashMap::<String, String>::new();
+        let mut report_paths = Vec::<String>::new();
+
+        let primary_purls = match self.service.get_primary_purls().await {
+            Ok(Some(purls)) => {
+                for purl in purls.clone() {
+                    format!("A PURL: {}", purl);
+                }
+                purls
+            },
+            Ok(None) => {
+                format!("NO PURL: No error, but no result");
+                Vec::<String>::new()
```

Comment:
We should write that error to MongoDB, but this will simply run the analytic with no purls if there is no result. Same as stopping here.  I'll take a look at what you did and see what that is.

Reply:
OK, the task will fail now.

Reply:
OK, changed them to panic!() macros, I hope that is what you think is appropriate by marking those errors ad ''unrecoverable'.  I'm no longer returning errors here now, if this still isn't correct then I'm going to need more instruction unfortunately.

Reply:
Went back to returning errors in all error cases